be/src/storage/lake/gc.cpp

```diff
@@ -146,10 +209,35 @@ static Status delete_tablet_metadata(std::string_view root_location, const std::
                     continue;
                 }
             }
-            auto path = join_path(metadata_root_location, tablet_metadata_filename(tablet_id, versions[i]));
-            LOG(INFO) << "Deleting " << path;
-            auto st = fs->delete_file(path);
-            LOG_IF(WARNING, !st.ok() && !st.is_not_found()) << "Fail to delete " << path << ": " << st;
+            // 1. get tablet meta pair
+            auto meta_pair = get_meta_pair(tablet_mgr, metadata_root_location, tablet_id, versions[i], versions[i + 1]);
+            if (!meta_pair.ok()) {
+                LOG(ERROR) << "get meta pair failed " << meta_pair.status().to_string() << " tablet_id: " << tablet_id;
+                continue;
+            }
+            if ((*meta_pair).first->schema().keys_type() == KeysType::PRIMARY_KEYS) {
```

Comment:
All tablets in the same directory(root_location) have the same schema, so you only need to check the schema once outside the loop.